clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -182,29 +180,35 @@ object UncommittedGarbageCollector {
           .repartition(dataDF.col("address"))
           .except(committedDF)
           .except(uncommittedDF)
+          .cache()
+
+        committedDF.unpersist()
```

Comment:
I tried to find the documentation for this but it wasn't very clear.
What exactly does unpersist do - and what are the performance implications for this change

Reply:
Release memory associated with the dataframe - in this code I assume that from this point I'll go over the `addressesToDelete` cached results and do not process committed and uncommitted dataframe again.
In the we would had to process these data frame, it would run the processing again - which will be very bad for performance.